src/microbe_stage/ColonyCompoundBag.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+[UseThriveSerializer]
+public class ColonyCompoundBag : ICompoundStorage
+{
+    public ColonyCompoundBag()
+    {
+    }
+
+    public ColonyCompoundBag(MicrobeColony colony)
+    {
+        Colony = colony;
+    }
+
+    [JsonProperty]
+    public MicrobeColony Colony { get; set; }
+
+    public float Capacity => GetCompoundBags().Sum(p => p.Capacity);
+
+    /// <summary>
+    ///   Distributes the amount above average compounds to the cells below average compounds.
+    /// </summary>
+    public void DistributeCompoundSurplus()
+    {
+        var bags = GetCompoundBags().ToList();
+        using var compounds = GetEnumerator();
+
+        while (compounds.MoveNext())
+        {
+            var compound = compounds.Current;
+            if (!compound.Key.CanBeDistributed)
+                continue;
+
+            var average = compound.Value / bags.Count;
+
+            foreach (var bag in bags)
+            {
+                var surplus = bag.GetCompoundAmount(compound.Key) - average;
+                if (surplus > 0)
+                    bag.TakeCompound(compound.Key, surplus);
+                else
+                    bag.AddCompound(compound.Key, -surplus);
+            }
+        }
+    }
+
+    public IEnumerator<KeyValuePair<Compound, float>> GetEnumerator()
+    {
+        return GetCompoundBags()
+            .SelectMany(p => p.Compounds)
+            .GroupBy(p => p.Key)
+            .Select(p => new KeyValuePair<Compound, float>(p.Key, p.Sum(x => x.Value)))
+            .GetEnumerator();
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
```

Comment:
this can't have an access control modifier, because it is an explicit interface implementation. The `CompoundBag` in the master also doesn't have an access modifier

Reply:
Ah okay, interesting. I was kinda suspecting that (because I would expect CI to complain), but decided to comment anyway.